flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -1246,7 +1246,46 @@ public long count() throws Exception {
 		final TypeInformation<K> keyType = TypeExtractor.getKeySelectorTypes(keyExtractor, getType());
 		return new PartitionOperator<T>(this, PartitionMethod.HASH, new Keys.SelectorFunctionKeys<T, K>(clean(keyExtractor), this.getType(), keyType), Utils.getCallLocationName());
 	}
-	
+
+	/**
+	 * Range-partitions a DataSet on the specified key fields.
+	 * <p>
+	 * <b>Important:</b>This operation shuffles the whole DataSet over the network and can take significant amount of time.
```

Comment:
We should add that range partitioning requires an additional pass over the data to obtain the distribution.